app/controllers/concerns/listings_toolkit.rb

```diff
@@ -120,11 +98,28 @@ def update
     process_after_update
   end
 
+  private
+
+  def set_and_authorize_listing
```

Comment:
Every action that uses `set_listing` calls `authorize @listing` as its first line so I added that to the action.